packages/back-end/src/services/datasource.ts

```diff
@@ -16,10 +16,12 @@ import {
   DataSourceParams,
   DataSourceSettings,
   DataSourceType,
+  ExposureQuery,
 } from "../../types/datasource";
 import Mysql from "../integrations/Mysql";
 import Mssql from "../integrations/Mssql";
 import { getDataSourceById } from "../models/DataSourceModel";
+import * as thisModule from "./datasource"; //needed for jest.mock
```

Comment:
A bit unhappy that I needed to do this ... seems a bit hacky.  But this was the only way I could get the mock of getSourceIntegrationObject from the test file to be used within this file.  Without it, it uses the real function.  I also tried out "rewire" npm package which works without this ... but it needs to reference the built file, which means yarn build would have to be run before the tests work which is not good for local development.

Reply:
Can you just change the function to accept an integration object instead of a datasource object? Then you don't need to mock anything.

Reply:
yes that I suppose makes it easier.